content/company/strategy/code-graph/index.md

```diff
@@ -51,6 +51,8 @@ We also set our goals around these themes, you can see current FY22-Q4 goals as
 
 Learning what's possible and how to get the most out of our product tends to be trial and error, especially for an individual developer who might have a unique set of use cases and interests for which they would benefit from exploring. Making onboarding and interaction easier also helps our users and sales teams demonstrate the value of our product more quickly than is possible today.
 
+As the market matures, it’s likely we will face competition from companies that vertically specialise in a language ecosystem. One mode of competition that we see from Moderne (for example) is to provide large number of recipes to auto-fix code. While we likely won’t focus on maintaining an exhaustive collection of recipes for a given language, we could create some core examples for each language until they become sufficiently good to drive community involvement.
```

Comment:
```suggestion
As the market matures, it’s likely we will face competition from companies that vertically specialise in a language ecosystem. It will be important to monitor these to ensure we understand the key innovations, bringing them in (and generalizing them to all languages where possible.) We want to do all languages well, and differentiate on supporting all the languages you use. We don't want to do find ourselves in a situation where we are doing all languages, but doing them poorly.
```